### ISSUE TITLE
The active style and active class of a Link component are applied only if the URL matches exactly

### ISSUE THREAD

**Issue by nikoladev** [gatsbyjs/gatsby#8160]
## Description

The `activeStyle` and `activeClassName` props of the `Link` component are not applied if the URL has a trailing slash (or if it lacks a trailing slash, depending on the value of the `to` prop).

Basically, if a user navigates to `site.com/example/` and then to `site.com/example`, only one of those URLs will have correct styling applied to the `Link` component.

### Steps to reproduce

Create a new project:

```bash
gatsby new gatsby-link-test https://github.com/gatsbyjs/gatsby-starter-default#v2

cd gatsby-link-test
```

Replace all the code in `src/components/header.js` with this:

```jsx
import React from 'react'
import { Link } from 'gatsby'

const style = {
  color: 'white',
  textDecoration: 'none',
}

const activeStyle = {
  color: 'red',
}

const Header = ({ siteTitle }) => (
  <div
    style={{
      background: 'rebeccapurple',
      marginBottom: '1.45rem',
    }}
  >
    <div
      style={{
        margin: '0 auto',
        maxWidth: 960,
        padding: '1.45rem 1.0875rem',
      }}
    >
      <h1 style={{ margin: 0 }}>
        <Link
          to="/"
          style={style}
          activeStyle={activeStyle}
        >
          {siteTitle}
        </Link>
        {' '}
        <Link
          to='/page-2/' // <-------------- Try switching these around
          // to='/page-2' // <-------------- Try switching these around
          style={style}
          activeStyle={activeStyle}
        >
          Page 2
        </Link>
      </h1>
    </div>
  </div>
)

export default Header
```

Start a development server:

```bash
gatsby develop
```

Go to `http://localhost:8000/page-2/` and the active link will be styled correctly. But if you go to `http://localhost:8000/page-2` (same link, but without the trailing slash) the `activeStyle` is no longer applied.

If you set `to='/page-2'` in `header.js` then the same problem occurs, only in reverse.

This all applies to `activeClassName` as well.

### Expected result

The styling of an active link should apply on `site.com/example` as well as on `site.com/example/`.

### Actual result

Only the exact URL gets the active style applied.

### Environment

```
System:
  OS: macOS High Sierra 10.13.6
  CPU: x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
  Shell: 5.3 - /bin/zsh
Binaries:
  Node: 8.12.0 - /usr/local/bin/node
  Yarn: 1.9.4 - /usr/local/bin/yarn
  npm: 6.4.1 - ~/.npm-global/bin/npm
Browsers:
  Chrome: 69.0.3497.92
  Firefox: 62.0
  Safari: 11.1.2
npmPackages:
  gatsby: next => 2.0.0-rc.24 
  gatsby-plugin-manifest: next => 2.0.2-rc.1 
  gatsby-plugin-offline: next => 2.0.0-rc.7 
  gatsby-plugin-react-helmet: next => 3.0.0-rc.1 
```

**Comment by stefanprobst** [comment]
Related: #7737
What you could do is use your own `getProps` on `<Link />`, see the [docs for reach-router](https://reach.tech/router/api/Link).

**Comment by kakadiadarpan** [comment]
Closed in favor of #7737
